resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnRMClient.scala

```diff
@@ -83,7 +83,7 @@ private[spark] class YarnRMClient extends Logging {
       localResources: Map[String, LocalResource]): YarnAllocator = {
     require(registered, "Must register AM before creating allocator.")
     new YarnAllocator(driverUrl, driverRef, conf, sparkConf, amClient, appAttemptId, securityMgr,
-      localResources, new SparkRackResolver())
+      localResources, new SparkRackResolver(conf))
```

Comment:
Do you want the shared instance here instead?

Reply:
oh good point, i do want the shared instance, to use a shared `CachedDnsToSwitchMapping`.  In cluster mode that will make sure we re-use the cache.